main.go

```diff
@@ -646,9 +660,23 @@ func main() {
 		buildTestParams.CleanBuild = configs.IsCleanBuild
 	}
 
+	if simulatorDebug != never {
+		log.Infof("Enabling Simulator verbose log for better diagnostics")
+		// Boot the simulator now, so verbose logging can be enabled and it is kept booted after running tests,
+		// this helps to collect more detailed debug info
+		if err := simulatorBoot(sim.ID); err != nil {
```

Comment:
is it possible we try to boot the simulator some cases twice? (simulatorDebug != never && sim.Status == simulatorShutdownState && !configs.HeadlessMode, line 682)

Reply:
In case if it is started it will not start again. The headless mode seems not useful other than to show the actual simulator UI and could be removed IMO.